apps/src/templates/manageStudents/ManageStudentsSharingCell.jsx

```diff
@@ -0,0 +1,65 @@
+import React, {Component, PropTypes} from 'react';
+import {connect} from 'react-redux';
+import {editStudent} from './manageStudentsRedux';
+import {Checkbox} from 'react-bootstrap';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+class ManageStudentsSharingCell extends Component {
+  static propTypes = {
+    id: PropTypes.number.isRequired,
+    isEditing: PropTypes.bool,
+    disabled: PropTypes.bool,
+    checked: PropTypes.bool,
+    //Provided by redux
+    editStudent: PropTypes.func,
+  };
+
+  state = {
+    checked: this.props.checked
+  };
+
+  toggleSharingCheckbox = () => {
+    this.setState({
+      checked: !this.state.checked
+    }, this.changeSharing);
+  };
+
+  changeSharing = (e) => {
+    this.props.editStudent(this.props.id, {sharingDisabled: !this.state.checked});
+  };
+
+  render() {
+    const {disabled, checked} = this.props;
+    return (
+      <div>
+        {!this.props.isEditing &&
+          <div>
+            {checked &&
+              <FontAwesome
+                icon="check"
+                className="fa-check"
+              />
+            }
+          </div>
+        }
+        {this.props.isEditing &&
+          <div>
+            <Checkbox
+              disabled={disabled}
+              checked={this.state.checked}
```

Comment:
internal state here is redundant since you're also saving it to redux.

`checked={this.props.checked}`
`onChange=this.changeSharing`

then changeSharing can get the value of the checkbox by doing something like `e.target.value`

Reply:
actually, that might give you the opposite boolean you'd need for redux, but you should be able to flip it.

Reply:
I used to state for checked so that the user can toggle it as many times as they want, and it will always check or uncheck based on their clicks. Then, just send to Redux whatever the state is on save. If I set `checked={this.props.checked}` whether the box is checked or not won't change - it will stay whatever it was initially.

Reply:
Based on the code you have here, you currently call `editStudent` every time you toggle the checkbox so redux gets the updates to the share setting on every toggle, not just on save, which looks correct. I think you probably just have a couple lines off that makes it not work when you make the change.
The name cell is a good example. You need to pass in both the `checked` value, and the `editedCheckedValue`. `checked` is used for display and `editedCheckedValue` is used for the value of the check box when editing: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/manageStudents/ManageStudentsNameCell.jsx#L24

We want `editingData` in redux to always reflect the edited values, so that we could call save or cancel from anywhere and it'd have the most up to date information.

Reply:
🤔I think it does work when I make the change.
![editing-data-updates](https://user-images.githubusercontent.com/12300669/37615081-a48cbac2-2b69-11e8-9d33-ecbc537fd2ea.gif) Should I be looking somewhere else to confirm?